cmake fails on OS X

```
[cmake] -- The C compiler identification is GNU 4.9.2
[cmake] -- The CXX compiler identification is GNU 4.9.2
[cmake] CMake Error at Modules/Platform/Darwin.cmake:76 (message):
[cmake]   CMAKE_OSX_DEPLOYMENT_TARGET is '10.10' but CMAKE_OSX_SYSROOT:
[cmake] 
[cmake]    ""
[cmake] 
[cmake]   is not set to a MacOSX SDK with a recognized version.  Either set
[cmake]   CMAKE_OSX_SYSROOT to a valid SDK or set CMAKE_OSX_DEPLOYMENT_TARGET to
[cmake]   empty.
[cmake] Call Stack (most recent call first):
[cmake]   Modules/CMakeSystemSpecificInformation.cmake:36 (include)
[cmake]   CMakeLists.txt:16 (project)
[cmake] 
[cmake] 
```

Applying the following patch:
```diff
diff --git a/pkgs/cmake.yaml b/pkgs/cmake.yaml
index 4da3973..4d77fd8 100644
--- a/pkgs/cmake.yaml
+++ b/pkgs/cmake.yaml
@@ -5,6 +5,15 @@ sources:
   url: http://www.cmake.org/files/v3.2/cmake-3.2.0-rc1.tar.gz
 
 build_stages:
+- when: platform == 'Darwin'
+  name: fix_deployment_target
+  after: prologue
+  before: configure
+  handler: bash
+  bash: |
+    #export CMAKE_OSX_DEPLOYMENT_TARGET=
+    unset MACOSX_DEPLOYMENT_TARGET
+
 - name: configure
   extra: ['--parallel=${HASHDIST_CPU_COUNT}']
```
it fails later, with:
```
[cmake] [ 64%] Building C object Utilities/cmcurl/lib/CMakeFiles/cmcurl.dir/vtls/cyassl.c.o
[cmake] [ 64%] Building C object Utilities/cmcurl/lib/CMakeFiles/cmcurl.dir/vtls/curl_schannel.c.o
[cmake] [ 65%] Building C object Utilities/cmcurl/lib/CMakeFiles/cmcurl.dir/vtls/curl_darwinssl.c.o
[cmake] [ 65%] Building C object Utilities/cmcurl/lib/CMakeFiles/cmcurl.dir/vtls/gskit.c.o
[cmake] /Users/certik/.hashdist/tmp/cmake-6l5ocd3laip5/Utilities/cmcurl/lib/vtls/curl_darwinssl.c: In function 'darwinssl_connect_step3':
[cmake] /Users/certik/.hashdist/tmp/cmake-6l5ocd3laip5/Utilities/cmcurl/lib/vtls/curl_darwinssl.c:2003:6: error: 'SecTrustEvaluateAsync' undeclared (first use in this function)
[cmake]    if(SecTrustEvaluateAsync != NULL) {
[cmake]       ^
[cmake] /Users/certik/.hashdist/tmp/cmake-6l5ocd3laip5/Utilities/cmcurl/lib/vtls/curl_darwinssl.c:2003:6: note: each undeclared identifier is reported only once for each function it appears in
[cmake] make[2]: *** [Utilities/cmcurl/lib/CMakeFiles/cmcurl.dir/vtls/curl_darwinssl.c.o] Error 1
[cmake] make[1]: *** [Utilities/cmcurl/lib/CMakeFiles/cmcurl.dir/all] Error 2
[cmake] make: *** [all] Error 2
```

--- a/base/cmake_package.py
+++ b/base/cmake_package.py
@@ -34,6 +34,9 @@ def configure(ctx, stage_args):
     else:
         conf_lines.append('-DCMAKE_BUILD_TYPE:STRING=Release')
 
+    if ctx.parameters['platform'] == 'Darwin':
+        conf_lines.append('-DCMAKE_OSX_DEPLOYMENT_TARGET:STRING=""')
+
     #cmake needs to be given all the dependency dirs as prefix paths
     #so that we search the hashdist directories before the system directories
     #CMake doesn't use the CPPFLAGS implicitly to find libraries

--- a/pkgs/cmake.yaml
+++ b/pkgs/cmake.yaml
@@ -8,6 +8,14 @@ sources:
   url: http://www.cmake.org/files/v3.2/cmake-3.2.2.tar.gz
 
 build_stages:
+- when: platform == 'Darwin'
+  name: fix_deployment_target
+  after: prologue
+  before: configure
+  handler: bash
+  bash: |
+    unset MACOSX_DEPLOYMENT_TARGET
+
 - name: configure
   extra: ['--parallel=${HASHDIST_CPU_COUNT}',
           '--system-bzip2',

--- a/pkgs/cmake.yaml
+++ b/pkgs/cmake.yaml
@@ -1,12 +1,24 @@
 extends: [autotools_package]
 
+dependencies:
+  build: [zlib, bzip2, curl]
+
 sources:
 - key: tar.gz:vxuu43rwaodxivs7flwyqzsbkrbuit5x
   url: http://www.cmake.org/files/v3.2/cmake-3.2.2.tar.gz
 
 build_stages:
 - name: configure
-  extra: ['--parallel=${HASHDIST_CPU_COUNT}']
+  extra: ['--parallel=${HASHDIST_CPU_COUNT}',
+          '--system-bzip2',
+          '--system-curl',
+          #'--system-expat',
+          #'--system-jsoncpp',
+          #'--system-libarchive',
+          '--system-zlib',
+          '--',
+          '-DCMAKE_PREFIX_PATH=${CURL_DIR}/lib;${ZLIB_DIR}/lib;${BZIP2_DIR}/lib',
+          ]
 
 when_build_dependency:
 - set: CMAKE

--- a/pkgs/cmake.yaml
+++ b/pkgs/cmake.yaml
@@ -1,8 +1,8 @@
 extends: [autotools_package]
 
 sources:
-- key: tar.gz:kcrzjtzo7wnjfxkbctv6fgmurgmezzxg
-  url: http://www.cmake.org/files/v3.2/cmake-3.2.0-rc1.tar.gz
+- key: tar.gz:vxuu43rwaodxivs7flwyqzsbkrbuit5x
+  url: http://www.cmake.org/files/v3.2/cmake-3.2.2.tar.gz
 
 build_stages:
 - name: configure

--- a/pkgs/cmake.yaml
+++ b/pkgs/cmake.yaml
@@ -1,7 +1,7 @@
 extends: [autotools_package]
 
 dependencies:
-  build: [zlib, bzip2, curl]
+  build: [zlib, bzip2, curl, openssl, libidn]
 
 sources:
 - key: tar.gz:vxuu43rwaodxivs7flwyqzsbkrbuit5x
@@ -26,6 +26,9 @@ build_stages:
           '--system-zlib',
           '--',
           '-DCMAKE_PREFIX_PATH=${CURL_DIR}/lib;${ZLIB_DIR}/lib;${BZIP2_DIR}/lib',
+          '-DCURL_INCLUDE_DIR=${CURL_DIR}/include',
+          '-DZLIB_INCLUDE_DIR=${ZLIB_DIR}/include',
+          '-DBZIP2_INCLUDE_DIR=${BZIP2_DIR}/include',
           ]
 
 when_build_dependency:

--- a/pkgs/cmake.yaml
+++ b/pkgs/cmake.yaml
@@ -1,7 +1,7 @@
 extends: [autotools_package]
 
 dependencies:
-  build: [zlib, bzip2, curl, openssl, libidn]
+  build: [zlib, bzip2, curl, openssl, libidn, expat]
 
 sources:
 - key: tar.gz:vxuu43rwaodxivs7flwyqzsbkrbuit5x
@@ -20,15 +20,16 @@ build_stages:
   extra: ['--parallel=${HASHDIST_CPU_COUNT}',
           '--system-bzip2',
           '--system-curl',
-          #'--system-expat',
+          '--system-expat',
           #'--system-jsoncpp',
           #'--system-libarchive',
           '--system-zlib',
           '--',
-          '-DCMAKE_PREFIX_PATH=${CURL_DIR}/lib;${ZLIB_DIR}/lib;${BZIP2_DIR}/lib',
+          '-DCMAKE_PREFIX_PATH=${CURL_DIR}/lib;${ZLIB_DIR}/lib;${BZIP2_DIR}/lib;${EXPAT_DIR}/lib',
           '-DCURL_INCLUDE_DIR=${CURL_DIR}/include',
           '-DZLIB_INCLUDE_DIR=${ZLIB_DIR}/include',
           '-DBZIP2_INCLUDE_DIR=${BZIP2_DIR}/include',
+          '-DEXPAT_INCLUDE_DIR=${EXPAT_DIR}/include',
           ]
 
 when_build_dependency:

